File: employee-system-api/src/main/java/com/dailycode/employee/service/EmployeeService.java
package com.dailycode.employee.service;

import com.dailycode.employee.model.Employee;

import java.util.List;

public interface EmployeeService {
    Employee createEmployee(Employee employee);

    List<Employee> getAllEmployee();

    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    void updateEmployee(Employee employee);
}

File: employee-system-ui/src/service/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = "http://localhost:8080/api/v1/employees";

class EmployeeService{

    //저장
    saveEmployee(employee){
        return axios.post(EMPLOYEE_API_BASE_URL,employee);
    }
    //리스트
    getEmployee(){
        return axios.get(EMPLOYEE_API_BASE_URL);
    }
    //삭제
    deleteEmployee(id){
        return axios.delete(EMPLOYEE_API_BASE_URL+"/"+id);
    }

    //수정 화면
    getEmployeeById(id){
        return axios.get(EMPLOYEE_API_BASE_URL+"/"+id);
    }
    //수정
    updateEmployee(employee){
        return axios.put(EMPLOYEE_API_BASE_URL,employee);
    }
}

export default new EmployeeService();